src/interp.rs

```diff
@@ -441,48 +275,24 @@ mod specialized {
 
             output
         }
-    }
 
-    impl<F> super::Interpolator for Interp4F<F>
-    where
-        F: Frame + std::fmt::Debug + Default + AsRef<[f32]>,
-    {
-        fn process(&mut self, src: &[f32], dst: &mut [f32]) {
-            assert_eq!(0, src.len() % F::CHANNELS);
-            assert_eq!(src.len() * FACTOR4, dst.len());
-            let frames = src.len() / F::CHANNELS;
-
-            for i in 0..frames {
-                let res = self.push(&F::from_planar(&src[i..], frames));
-                for c in 0..F::CHANNELS {
-                    for (f, frame) in res.iter().enumerate() {
-                        dst[c * frames * FACTOR4 + i * FACTOR4 + f] = frame.as_ref()[c];
-                    }
-                }
-            }
-        }
-
-        fn reset(&mut self) {
+        pub(crate) fn reset(&mut self) {
             self.buffer = Default::default();
         }
-
-        fn get_factor(&self) -> usize {
-            4
-        }
     }
 
-    #[derive(Debug)]
-    pub(super) struct Interp2F<F: Frame> {
+    #[derive(Debug, Clone)]
```

Comment:
OOC, why do we need `Clone` here?

Reply:
Since the "generic" true-peak analyzer creates a vector array of Interpolators. The vector-creation involves cloneing a value to fill the size of the vector.

Reply:
Ah, makes sense. Thanks!